src/test/java/mirror/UpdateTreeDiffInboundOnlyTest.java

```diff
@@ -0,0 +1,555 @@
+package mirror;
+
+import com.google.protobuf.ByteString;
+import mirror.UpdateTree.Node;
+import mirror.UpdateTreeDiff.DiffResults;
+import org.jooq.lambda.Seq;
+import org.junit.Test;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.jooq.lambda.Seq.seq;
+
+public class UpdateTreeDiffInboundOnlyTest {
```

Comment:
Wow, thanks for these tests.

Reply:
...that said, its fairly hard to tell what did/did not change between the regular / inbound / and outbound test cases.

I wonder if `UpdateTreeDiffInboundOnlyTest` could extend `UpdateTreeDiffTest` and then override the methods that are specifically different?

I've never really done that before (have one unit test extend another and selectively override tests) but I think? it should work.

Reply:
I wasn't really sure which properties I was expecting to hold as I did this, so I was using this as a reference more than anything else. That said, you're right about trying to inherit from UpdateTreeDiffTest.